packages/core/src/Machine.ts

```diff
@@ -93,11 +93,18 @@ export function createMachine<
     TServiceMap,
     TTypesMeta
   >,
-  options?: InternalMachineOptions<
-    TContext,
-    TEvent,
-    ResolveTypegenMeta<TTypesMeta, TEvent, BaseActionObject, TServiceMap>
-  >
+  options?: TTypesMeta extends infer EvaluatedTypesMeta
```

Comment:
This somehow forces TS to "evaluate" the final~ type for `TTypesMeta` quicker. My hunch is that it couldn't be discovered soon enough without typegen because `tsTypes` is an optional property, so it was missing then TS tried to "explore" the stuff further instead of using the fallback (`TypegenDisabled` default) immediately and then it couldn't compute stuff further down the road.

By doing it we kinda force it to utilize that `TypegenDisabled` default and it can proceed computation with that additional knowledge and that in turns allow for other stuff to be computed correctly.